### PR TITLE
fix: loading lazy messages when init app and simplify i18n options types

### PR DESCRIPTION
**Pull request by mario-mui**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/neat-planes-remain.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/core-app-api': patch
+---
+
+fix: loading lazy messages when init app and simplify i18n options types
```
Comment (Rugvip) [inline .changeset/neat-planes-remain.md:5]: Let's skip changesets for the followup changes for now, until i18n has been released in main line

**Review by Rugvip — CHANGES_REQUESTED** [review]
Alright!

Could you provide a quick description of the reason for the change as well?

File: .changeset/neat-planes-remain.md (hunk 1 above)
Comment (mario-mui) [reply .changeset/neat-planes-remain.md]: 1

**Comment by mario-mui** [comment]
> Alright!
> 
> Could you provide a quick description of the reason for the change as well?

Alright..

1. for types: 
```

```

File: packages/core-app-api/src/app/TranslationResource.ts [hunk 2]
```diff
@@ -18,13 +18,13 @@ import { TranslationRef } from '@backstage/core-plugin-api/alpha';
 
 /** @alpha */
 export type TranslationMessages<T> = T extends TranslationRef<infer R>
-  ? Record<string, Partial<R>>
+  ? Partial<R>
```
Comment (mario-mui) [inline packages/core-app-api/src/app/TranslationResource.ts:21]: for types, last version `createTranslationResource` with `lazyMessages` options need to be 
```ts
createTranslationResource({
    lazyMessages: {
        zh: () => Promise.resolve({ messages: { zh: { language: '语言' }}})
    }
})
```
due to `TranslationMessages ` wrap `Record` to specify language key

File: packages/core-app-api/src/apis/implementations/AppTranslationApi/AppTranslationImpl.ts [hunk 3]
```diff
@@ -136,18 +139,13 @@ export class AppTranslationApiImpl implements AppTranslationApi {
...
-    ) as string[];
-
-    Promise.allSettled(
-      [...fallbackLanguages, currentLanguage].map(addLanguage),
-    ).then(results => {
-      if (results.some(result => result.status === 'fulfilled')) {
-        this.i18n.emit('loaded');
-      }
-    });
+    Promise.allSettled((options.supportedLngs || []).map(addLanguage)).then(
```
Comment (mario-mui) [inline packages/core-app-api/src/apis/implementations/AppTranslationApi/AppTranslationImpl.ts:142]: supportLanguages maybe `['zh', 'en']` and `fallbackLanguages` just to be 'en'. When I add `zh` resources in 'createApp', need to add zh resource to i18next context

**Comment by mario-mui** [comment]
> Alright!
> 
> Could you provide a quick description of the reason for the change as well?

sure, left so message by code review

**Review by Rugvip — APPROVED** [review]
Alright, let's :shipit:!

**Pull request merged**